src/sender.py

```
@@ -311,9 +398,14 @@ def run(self):
                     
                     if ret_code                             !=  200:
                         t.debug_print(f"Failed to send data (code {ret_code}).", self.name)
+                        queue.requeue(payload)
                         time.sleep(1)
+                    else:
+                        queue.ack(payload)
             except Exception as e:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)